kafka/adminapi.go

```diff
@@ -3516,6 +3592,74 @@ func (a *AdminClient) DeleteRecords(ctx context.Context,
 	return result, nil
 }
 
+// ElectLeaders performs Preferred or Unclean Elections for the specified topic Partitions.
+//
+// Parameters:
+//   - `ctx` - context with the maximum amount of time to block, or nil for
+//     indefinite.
+//   - `electLeaderRequest` - ElectLeadersRequest containing the election type
+//     and the partitions to elect leaders for.
+//   - `options` - ElectLeadersAdminOption options.
+//
+// Returns ElectLeadersResult, which contains a slice of TopicPartitions containing the partitions for which the leader election was performed.
+// If we are passing partitions as nil, the broker will perform leader elections for all partitions,
+// but the results will only contain partitions for which there was an election or resulted in an error.
+// Individual TopicPartitions inside the ElectLeadersResult should be checked for errors.
+// Additionally, an error that is not nil for client-level errors is returned.
+func (a *AdminClient) ElectLeaders(ctx context.Context, electLeaderRequest ElectLeadersRequest, options ...ElectLeadersAdminOption) (result ElectLeadersResult, err error) {
+
+	err = a.verifyClient()
+	if err != nil {
+		return result, err
+	}
+
+	cTopicPartitions := newCPartsFromTopicPartitions(electLeaderRequest.partitions)
+	defer C.rd_kafka_topic_partition_list_destroy(cTopicPartitions)
+
+	cElectLeadersRequest := C.rd_kafka_ElectLeaders_new(C.rd_kafka_ElectionType_t(electLeaderRequest.electionType), cTopicPartitions)
+	defer C.rd_kafka_ElectLeaders_destroy(cElectLeadersRequest)
```

Comment:
This way we handle the nil case differently from the empty case
```suggestion
        var cTopicPartitions *C.rd_kafka_topic_partition_list_t
	if electLeaderRequest.partitions != nil {
		cTopicPartitions = newCPartsFromTopicPartitions(electLeaderRequest.partitions)
		defer C.rd_kafka_topic_partition_list_destroy(cTopicPartitions)
	}
```